### PR TITLE
Moved Table.explode to IR.

### PR DESCRIPTION
**Pull request by cseed**

### CODE REVIEW

File: src/main/scala/is/hail/expr/Relational.scala [hunk 1]
```diff
@@ -1131,3 +1131,59 @@ case class TableMapGlobals(child: TableIR, newRow: IR) extends TableIR {
     TableValue(typ, newGlobals, tv.rvd)
   }
 }
+
+case class TableExplode(child: TableIR, column: String) extends TableIR {
+  def children: IndexedSeq[BaseIR] = Array(child)
+
+  private val columnIdx = child.typ.rowType.fieldIdx(column)
+  private val columnType = child.typ.rowType.types(columnIdx).asInstanceOf[TContainer]
```
Comment (jbloom22) [inline src/main/scala/is/hail/expr/Relational.scala]: check that columnType is a container type in Python?

File: src/main/scala/is/hail/expr/Relational.scala [hunk 2]
```diff
@@ -1131,3 +1131,59 @@ case class TableMapGlobals(child: TableIR, newRow: IR) extends TableIR {
...
+        it.flatMap { rv =>
+          val isMissing = isMissingF()(rv.region, rv.offset, false)
+          if (isMissing)
+            Iterator.empty
+          else {
+            val end = rv.region.size
+            val n = lengthF()(rv.region, rv.offset, false)
+            Iterator.range(0, n)
+              .map { i =>
+                // rv.region.clear(end)
```
Comment (jbloom22) [inline src/main/scala/is/hail/expr/Relational.scala]: commented code

File: src/main/scala/is/hail/expr/Relational.scala (hunk 1 above)
Comment (cseed) [reply src/main/scala/is/hail/expr/Relational.scala]: Good catch, I thought that was happening.

**Comment by cseed** [comment]
I changed is_container to match the meaning in Scala (is array, set or dict) and added is_compound to have the old meaning (type that can contain other types).

**Pull request merged**